bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/ImageLoader.java

```diff
@@ -149,10 +149,71 @@ void reset() {
  * </ul>
  */
 public ImageData[] load(InputStream stream) {
+	return loadDefault(stream);
+}
+
+/**
+ * Loads an array of <code>ImageData</code> objects from the
+ * specified input stream. If the stream is a SVG File and zoom is not 0,
+ * this method will try to rasterize the SVG.
+ * Throws an error if either an error occurs while loading the images, or if the images are not
+ * of a supported type. Returns the loaded image data array.
+ *
+ * @param stream the input stream to load the images from
+ * @param zoom the zoom factor to apply when rasterizing a SVG.
+ * A value of 0 means that the standard method for loading should be used.
+ * @return an array of <code>ImageData</code> objects loaded from the specified input stream
+ *
+ * @exception IllegalArgumentException <ul>
+ *    <li>ERROR_NULL_ARGUMENT - if the stream is null</li>
+ * </ul>
+ * @exception SWTException <ul>
+ *    <li>ERROR_IO - if an IO error occurs while reading from the stream</li>
+ *    <li>ERROR_INVALID_IMAGE - if the image stream contains invalid data</li>
+ *    <li>ERROR_UNSUPPORTED_FORMAT - if the image stream contains an unrecognized format</li>
+ * </ul>
+ *
+ * @since 3.129
+ */
+public ImageData[] load(InputStream stream, int zoom) {
+	if (stream == null) SWT.error(SWT.ERROR_NULL_ARGUMENT);
+    reset();
+    byte[] bytes = null;
+	try {
+		bytes = stream.readAllBytes();
+	} catch (IOException e) {
+		SWT.error(SWT.ERROR_IO, e);
+	}
+	SVGRasterizer rasterizer = SVGRasterizerRegistry.getRasterizer();
+	if (rasterizer != null && zoom != 0) {
+		try (InputStream imageStream = new ByteArrayInputStream(bytes)) {
+			if (rasterizer.isSVGFile(imageStream)) {
+				float scalingFactor = zoom / 100.0f;
+				try (InputStream svgFileStream = new ByteArrayInputStream(bytes)) {
+					ImageData rasterizedData = rasterizer.rasterizeSVG(svgFileStream, scalingFactor);
+					if (rasterizedData != null) {
+						data = new ImageData[]{rasterizedData};
+					    return data;
+					}
+				}
+			}
+		} catch (IOException e) {
+			//ignore.
+		}
+	}
+	try (InputStream fallbackStream = new ByteArrayInputStream(bytes)) {
+		return loadDefault(fallbackStream);
+	} catch (IOException e) {
+		SWT.error(SWT.ERROR_IO, e);
+	}
+	return null;
```

Comment:
With https://github.com/eclipse-platform/eclipse.platform.swt/pull/1638#discussion_r1881143635 and https://github.com/eclipse-platform/eclipse.platform.swt/pull/1638#discussion_r1881114672, you could avoid reading the entire file-content into a byte-array and just wrap `stream` it into a `BufferedInputStream` if marking is not supported and assuming that the rasterizeSVG() will reset the stream if it has detected that the content isn't an SVG:
```suggestion
	SVGRasterizer rasterizer = SVGRasterizerRegistry.getRasterizer();
	if (rasterizer != null && zoom != 0) {
		if (!stream.markSupported()) {
			stream = new BufferedStream(stream);
		}
		float scalingFactor = zoom / 100.0f;
		Optional<ImageData> rasterizedData = rasterizer.rasterizeSVG(stream, scalingFactor);
		if (rasterizedData.isPresent()) {
			data = new ImageData[]{rasterizedData.get()};
			return data;
		}
	}
	try {
		return loadDefault(stream);
	} catch (IOException e) {
		SWT.error(SWT.ERROR_IO, e);
	}
	return null;
```

Reply:
This does not fix the problem that JSVG closes the InputStream as @laeubi explained [here](https://github.com/eclipse-platform/eclipse.platform.swt/pull/1638#discussion_r1881488576). I could use an `BufferedInputStream` for `isSVGFile()` but not for `rasterizeSVG()`.